core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/UniqueIDJDBCUserStoreManager.java

```diff
@@ -4823,32 +4829,15 @@ private void buildGroupWhereConditions(SqlBuilder sqlBuilder, String operation,
             sqlBuilder.where("R.UM_ROLE_NAME LIKE ?", "%" + value + "%");
         } else if (ExpressionOperation.SW.toString().equals(operation)) {
             sqlBuilder.where("R.UM_ROLE_NAME LIKE ?", value + "%");
+        } else if (ExpressionOperation.NE.toString().equals(operation)) {
+            sqlBuilder.where("(R.UM_ROLE_NAME IS NULL OR R.UM_ROLE_NAME <> ?)", value);
         }
     }
 
-    private void multiGroupMySqlQueryBuilder(SqlBuilder sqlBuilder, int groupFilterCount,
-            ExpressionCondition expressionCondition) {
-
-        if (groupFilterCount == 0) {
-            buildGroupWhereConditions(sqlBuilder, expressionCondition.getOperation(),
-                    expressionCondition.getAttributeValue());
-        } else {
-            buildGroupConditionWithOROperator(sqlBuilder, expressionCondition.getOperation(),
-                    expressionCondition.getAttributeValue());
-        }
-    }
-
-    private void buildGroupConditionWithOROperator(SqlBuilder sqlBuilder, String operation, String value) {
+    private void multiGroupMySqlQueryBuilder(SqlBuilder header, List<SqlBuilder> subSqlBuilders,
+                                             ExpressionCondition expressionCondition) {
 
-        if (ExpressionOperation.EQ.toString().equals(operation)) {
-            sqlBuilder.updateSqlWithOROperation("R.UM_ROLE_NAME = ?", value);
-        } else if (ExpressionOperation.EW.toString().equals(operation)) {
-            sqlBuilder.updateSqlWithOROperation("R.UM_ROLE_NAME LIKE ?", "%" + value);
-        } else if (ExpressionOperation.CO.toString().equals(operation)) {
-            sqlBuilder.updateSqlWithOROperation("R.UM_ROLE_NAME LIKE ?", "%" + value + "%");
-        } else if (ExpressionOperation.SW.toString().equals(operation)) {
-            sqlBuilder.updateSqlWithOROperation("R.UM_ROLE_NAME LIKE ?", value + "%");
-        }
+        subSqlBuilders.add(buildMySqlGroupSubSqlBuilder(header, expressionCondition));
```

Comment:
We may able move the logic of the method here instead of separate private method. 
If there are any common usages, let's use the private method approach.

Reply:
Resolved via https://github.com/wso2/carbon-kernel/pull/4305/commits/ac37534bb69427c5149b7558b89fd10120e62ad9